src/naf-dialog-adapter.js

```diff
@@ -613,141 +582,197 @@ export default class DialogAdapter {
-            dtlsParameters
-          })
-          .then(callback)
-          .catch(errback);
-      });
+        callback({ id });
+      } catch (error) {
+        this.emitRTCEvent("error", "Signaling", () => `[produce] error: ${error}`);
+        errback(error);
+      }
+    });
 
-      this._sendTransport.on("connectionstatechange", connectionState => {
-        let level = "info";
-        if (connectionState === "failed" || connectionState === "disconnected") {
-          level = "error";
-        }
-        this.emitRTCEvent(level, "RTC", () => `Send transport [connectionstatechange]: ${connectionState}`);
+    if (this._localMediaStream) {
+      this.createMissingProducers(this._localMediaStream);
+    }
+  }
 
-        this.checkSendIceStatus(connectionState);
-      });
+  async closeSendTransport() {
+    if (this._micProducer) {
+      this._micProducer.close();
+      this._protoo.connected && this._protoo.request("closeProducer", { producerId: this._micProducer.id });
+      this._micProducer = null;
+    }
 
-      this._sendTransport.on("produce", async ({ kind, rtpParameters, appData }, callback, errback) => {
-        this.emitRTCEvent("info", "RTC", () => `Send transport [produce]: ${kind}`);
-        try {
-          // eslint-disable-next-line no-shadow
-          const { id } = await this._protoo.request("produce", {
-            transportId: this._sendTransport.id,
-            kind,
-            rtpParameters,
-            appData
-          });
+    if (this._videoProducer) {
+      this._videoProducer.close();
+      this._protoo.connected && this._protoo.request("closeProducer", { producerId: this._videoProducer.id });
+      this._videoProducer = null;
+    }
 
-          callback({ id });
-        } catch (error) {
-          this.emitRTCEvent("error", "Signaling", () => `[produce] error: ${error}`);
-          errback(error);
-        }
-      });
+    if (!this._sendTransport?._closed) {
+      this._sendTransport.close();
+    }
 
-      // Create mediasoup Transport for sending (unless we don't want to consume).
-      const recvTransportInfo = await this._protoo.request("createWebRtcTransport", {
-        producing: false,
-        consuming: true,
-        sctpCapabilities: undefined
-      });
+    if (this._protoo.connected) {
+      try {
+        await this._protoo.request("closeWebRtcTransport", { transportId: this._sendTransport?.id });
+      } catch (err) {
+        error(err);
+      }
+    }
+  }
 
-      this._recvTransport = this._mediasoupDevice.createRecvTransport({
-        id: recvTransportInfo.id,
-        iceParameters: recvTransportInfo.iceParameters,
-        iceCandidates: recvTransportInfo.iceCandidates,
-        dtlsParameters: recvTransportInfo.dtlsParameters,
-        sctpParameters: recvTransportInfo.sctpParameters,
-        iceServers,
-        iceTransportPolicy: this._iceTransportPolicy
-      });
+  async createRecvTransport(iceServers) {
```

Comment:
Don't think we should do it in this PR but just noting for future cleanup. Seems like we could DRY up creating the send/receive transports. A lot of the code in here is the same for both.